governator-core/src/main/java/com/netflix/governator/conditional/ConditionalElementImpl.java

```diff
@@ -0,0 +1,84 @@
+package com.netflix.governator.conditional;
+
+import java.lang.annotation.Annotation;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import com.google.inject.Key;
+import com.google.inject.internal.Annotations;
+
+class ConditionalElementImpl implements ConditionalElement {
+    private static final AtomicInteger nextUniqueId = new AtomicInteger(1);
+
+    private final int uniqueId;
+    private final String keyName;
+    private final Conditional condition;
+
+    ConditionalElementImpl(String keyName, Conditional condition) {
+        this(keyName, condition, nextUniqueId.incrementAndGet());
+    }
+
+    ConditionalElementImpl(String keyName, Conditional condition, int uniqueId) {
+        this.uniqueId = uniqueId;
+        this.keyName = keyName;
+        this.condition = condition;
+    }
+
+    @Override
+    public String keyName() {
+        return keyName;
+    }
+
+    @Override
+    public int uniqueId() {
+        return uniqueId;
+    }
+
+    public Conditional getCondition() {
+        return condition;
+    }
+
+    @Override
+    public Class<? extends Annotation> annotationType() {
+        return ConditionalElement.class;
+    }
+
+    @Override
+    public String toString() {
+        return "@" + ConditionalElement.class.getName() + "(keyName=" + keyName
+                + ",uniqueId=" + uniqueId + ",condition=" + condition + ")";
+    }
+
+    @Override
+    public boolean equals(Object o) {
```

Comment:
seems flawed that Condition is not part of the equality check, but is part of the object state and passed in via constructor.  I would include at least an identity comparison for Condition here.